homeassistant/components/group/__init__.py

```diff
@@ -562,12 +567,29 @@ async def async_create_group(
             ENTITY_ID_FORMAT, object_id or name, hass=hass
         )
 
-        # If called before the platform async_setup is called (test cases)
-        if (component := hass.data.get(DOMAIN)) is None:
-            component = hass.data[DOMAIN] = EntityComponent(_LOGGER, DOMAIN, hass)
+        return group
+
+    @staticmethod
+    async def async_create_group(
+        hass,
+        name,
```

Comment:
Let's type new or changed core methods, if not a lot of extra work.

Reply:
Happy to do so.  I'm always not sure if I should as I don't want to generate more review work